### DISCUSSION TITLE
Re: winget should not install an already installed package

### DISCUSSION THREAD

**Discussion by denelon** [microsoft/winget-cli#1694] [Q&A]
Related to:
* #929

https://github.com/microsoft/winget-cli/issues/929#issuecomment-965729033

**Comment by jedieaston** [comment]
I wonder if upgrades running automatically on install is good default behavior. A user might just want to check if a package has already been installed, and if it hasn’t then install it. APT and yum work this way, although I might just be thinking “idempotently”. 

```
> winget install “Visual Studio Code”

Found Microsoft Visual Studio Code [Microsoft.VisualStudioCode]

This package is already installed, but an upgrade is available (v1.16.2). 

To upgrade, run “winget upgrade Microsoft.VisualStudioCode”

To force a reinstall, append “—force” to your previous command. 


> 

```

It may also be that there’s a reason the user is avoiding upgrading, and assuming that’s what they meant could be dangerous (what if an upgrade has breaking changes for their workflow, and they forgot the -v argument?).

**Reply by ghost** [reply]
I do not like this approach that i.e. Chocolatey has, something that you put forward here. It makes automation difficult (Intune/SCCM). If it does not install aleady installed packages, maybe add a parameter to winget install - something like: "winget install Microsoft.VisualStudioCode --allow-upgrades". It will then compare versions and upgrade if needed.
